iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -190,6 +180,78 @@ public void commitDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTable,
     }
   }
 
+  @Override
+  public void preAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, EnvironmentContext context)
+      throws MetaException {
+    HiveMetaHook.super.preAlterTable(hmsTable, context);
```

Comment:
any specific reason to do this?
Not wrong, just maybe we should do it everywhere or nowhere

Reply:
The default implementation of preAlterTable does some basic validation of the alter operation type. I thought this validation is useful.